pypesto/profile/profile.py

```diff
@@ -86,139 +88,37 @@ def create_next_guess(x, par_index, par_direction_, profile_options_,
     # existing list of profiles
     global_opt = initialize_profile(problem, result, result_index,
                                     profile_index, profile_list)
+    # if engine==None set SingleCoreEngine() as default
+    if engine is None:
+        engine = SingleCoreEngine()
 
-    # loop over parameters for profiling
+    # create Tasks
+    tasks = []
+    # loop over parameters to create tasks
     for i_par in profile_index:
         # only compute profiles for free parameters
         if i_par in problem.x_fixed_indices:
             continue
 
-        # create an instance of ProfilerResult, which will be appended to the
-        # result object, when this profile is finished
         current_profile = result.profile_result.get_profiler_result(
             i_par=i_par, profile_list=profile_list)
 
-        # compute profile in descending and ascending direction
-        for par_direction in [-1, 1]:
-            # flip profile
-            current_profile.flip_profile()
+        task = ProfilerTask(
+            current_profile=current_profile,
+            problem=problem,
+            optimizer=optimizer,
+            options=profile_options,
+            create_next_guess=create_next_guess,
+            global_opt=global_opt,
+            i_par=i_par
+        )
+        tasks.append(task)
 
-            # compute the current profile
-            current_profile = walk_along_profile(
-                current_profile=current_profile,
-                problem=problem,
-                par_direction=par_direction,
-                optimizer=optimizer,
-                options=profile_options,
-                create_next_guess=create_next_guess,
-                global_opt=global_opt,
-                i_par=i_par)
+    # execute the tasks with Engine
+    ret = engine.execute(tasks)
```

Comment:
```suggestion
    indexed_profiles = engine.execute(tasks)
```

To match the suggestion a few lines below.